freqtrade/optimize/backtesting.py

```diff
@@ -359,77 +359,121 @@ def _get_close_rate(self, sell_row: Tuple, trade: LocalTrade, sell: SellCheckTup
+            open_fee_rate = side_1 * trade.open_rate * (1 + side_1 * trade.fee_open)
+            close_rate = -side_1 * (roi_rate + open_fee_rate) / (trade.fee_close - side_1 * 1)
+            if is_short:
+                is_new_roi = sell_row[OPEN_IDX] < close_rate
+            else:
+                is_new_roi = sell_row[OPEN_IDX] > close_rate
+            if (trade_dur > 0 and trade_dur == roi_entry
+                    and roi_entry % self.timeframe_min == 0
+                    and is_new_roi):
+                # new ROI entry came into effect.
+                # use Open rate if open_rate > calculated sell rate
+                return sell_row[OPEN_IDX]
+
+            if (trade_dur == 0 and (
+                (
+                    is_short
+                    # Red candle (for longs), TODO: green candle (for shorts)
+                    and sell_row[OPEN_IDX] < sell_row[CLOSE_IDX]  # Red candle
+                    and trade.open_rate > sell_row[OPEN_IDX]  # trade-open below open_rate
+                    and close_rate < sell_row[CLOSE_IDX]
+                )
+                or
+                (
+                    not is_short
                     # Red candle (for longs), TODO: green candle (for shorts)
                     and sell_row[OPEN_IDX] > sell_row[CLOSE_IDX]  # Red candle
                     and trade.open_rate < sell_row[OPEN_IDX]  # trade-open below open_rate
                     and close_rate > sell_row[CLOSE_IDX]
-                ):
-                    # ROI on opening candles with custom pricing can only
-                    # trigger if the entry was at Open or lower.
-                    # details: https: // github.com/freqtrade/freqtrade/issues/6261
-                    # If open_rate is < open, only allow sells below the close on red candles.
-                    raise ValueError("Opening candle ROI on red candles.")
-                # Use the maximum between close_rate and low as we
-                # cannot sell outside of a candle.
-                # Applies when a new ROI setting comes in place and the whole candle is above that.
+                )
+            )):
+                # ROI on opening candles with custom pricing can only
+                # trigger if the entry was at Open or lower.
+                # details: https: // github.com/freqtrade/freqtrade/issues/6261
+                # If open_rate is < open, only allow sells below the close on red candles.
+                raise ValueError("Opening candle ROI on red candles.")
+
+            # Use the maximum between close_rate and low as we
+            # cannot sell outside of a candle.
+            # Applies when a new ROI setting comes in place and the whole candle is above that.
+            if is_short:
+                return max(min(close_rate, sell_row[HIGH_IDX]), sell_row[LOW_IDX])
+            else:
                 return min(max(close_rate, sell_row[LOW_IDX]), sell_row[HIGH_IDX])
```

Comment:
i don't think this condition is necessary.
Both are doing "min between close/high" - and "max between close/low" - just in the reverse sequence.

As we always assume that "high > low" (and i think that's a safe assumption) - they are actually behaving identically.